UI/backend/app.py

```diff
@@ -1,116 +1,131 @@
-    inst = data.get('instructor', '')
-    course = data.get('courseNameInst', '')
-    time  = data.get('timeSlotInst', '')
-
-
+    inst = data.get(UniJsonKeys.INSTRUCTOR, '')
+    course = data.get(UniJsonKeys.COURSE_NAME_INST, '')
+    time  = data.get(UniJsonKeys.TIME_SLOT_INST, '')
     if not inst:
-        return jsonify({'success': False, 'message': 'Instructor is required'}), 400
-    print(f"Received data: {data}")
-
-    app_dir = os.path.dirname(os.path.abspath(__file__))
-    grand_parent_dir = os.path.dirname(os.path.dirname(app_dir))
-    cpp_exec_path = os.path.join(grand_parent_dir, 'run.sh')
-
+        return jsonify({'success': False, 'message': 'Instructor is required'}), \
+            requests.codes.bad_request
+    insertFuncArgs = [inst]
+    if course:
+        insertFuncArgs.append(course)
+        insertCourseArgs = [course]
+        insertCourse(*insertCourseArgs)
+    if not course and time:
+        return jsonify({'success': False, 'message': 'If time is inputed course is mandatory'}), \
+            requests.codes.bad_request
+    if time:
+        tsProvided(time, insertFuncArgs)
     try:
-        if not course and not time:
-            result = subprocess.run(['bash', cpp_exec_path, '--addInstructor', inst], capture_output=True, text=True)
-        elif not time and course:
-            result = subprocess.run(['bash', cpp_exec_path, '--addInstructor', inst, course], capture_output=True, text=True)
-        elif time and course:
-            result = subprocess.run(['bash', cpp_exec_path, '--addInstructor', inst, course, time], capture_output=True, text=True)
-        else:
-            return jsonify({'success': False, 'message': 'If time is inputed course is mandatory'}), 400
-        if result.returncode == 0:
-            return jsonify({'success': True, 'message': 'Instructor added successfully!'})
-        else:
-            return jsonify({'success': False, 'message': 'Failed to add instructor', 'details': result.stderr}), 500
-    except Exception as e:
-        print(f"Exception occurred: {str(e)}")
-        return jsonify({'success': False, 'message': str(e)}), 500
+        insertInstructor(*insertFuncArgs)
+        return jsonify({'success': True, 'message': 'Instructor added successfully!'})
+    except Exception as dbError:
+        return jsonify({'success': False, 'message': str(dbError)}), \
+            requests.codes.internal_server_error
+    
+def replaceNoneWithEmptyString(d):
+    if isinstance(d, dict):
+        return {k: replaceNoneWithEmptyString(v) for k, v in d.items()}
+    elif isinstance(d, list):
+        return [replaceNoneWithEmptyString(item) for item in d]
+    else:
+        return '' if d is None else d
 
 @app.route('/schedule', methods=['GET'])
 def schedule():
-    app_dir = os.path.dirname(os.path.abspath(__file__))
-    grand_parent_dir = os.path.dirname(os.path.dirname(app_dir))
-    cpp_exec_path = os.path.join(grand_parent_dir, 'run.sh')
-
+    conn = connectDB()
```

Comment:
why do you create an object of DB connection? You have such object as global

Reply:
fixed